### ISSUE TITLE
MLG Water/Twisting Vines

### ISSUE THREAD

**Issue by lostsys** [N3ROO/Bettercolors#70]
**Is your feature request related to a problem? Please describe.**
No

**Describe the solution you'd like**
A realistic looking, auto mlg water/twisting vines (nether mlg water equivalent) that doesn't have major head snapping and or weird falling mechanics that could lead to weird packets being sent.

**Describe alternatives you've considered**
A toggle that will detect if you're mid air and if the player has a water bucket/twisting vines in hand, if so then if the player is looking at the block below them with such items, if all are true then it will place the water or twisting vines instead of trying to spam place them yourself.

**Additional context**
Just wanted to communicate a feature I've been thinking about for a while, honestly like the second option after thinking about it more but would like to know what you think, thanks!

**Comment by N3ROO** [comment]
Sounds cool!

Here is my idea (using pseudo-code):

```java
if (player.fallingAndWillTakeDamage()) {
    if (player.notLookingAtTheGround()) {
        player.assistAimingToTheGround(); // this can be an option, or we can set a limit (if the player is far from looking at the ground, then don't assist)
    }
    // here we are sure that the player is looking at the ground
    if (!player.holdingMlgItem()) {
        player.selectItem(); // water bucket or twisting vines
    }

    player.waitForBlockReach(whenCloseEnough = player.placeSelectedBlock);
}
```

Tell me what you think about it! :)

PS: I go on vacation this week, so I won't be available to implement this feature until the 2nd of September

**Comment by lostsys** [comment]
Yeah looks great and the option seems nice too, maybe something to do with pitch of the player, like if their pitch is around 30 (maybe lower or higher) it will assist to 90 to complete the mlg. Thanks again and have a nice vacation <3

**Comment by lolgamergotaids** [comment]
Yeah and like a auto ladder in bedwars

**Comment by N3ROO** [comment]
An auto ladder is not as simple as the "auto-water" and the "auto-vine" to look legit (since you have to place the ladder on the side of an other block), but I'll try to do something that looks legit, I can't promise anything

**Comment by lolgamergotaids** [comment]
Thx
